docs/OV_Runtime_UG/supported_plugins/GPU.md

```diff
@@ -258,7 +275,47 @@ The code snippet below demonstrates how to use dynamic batching in simple scenar
          :fragment: dynamic_batch
 
 
-For more details, see the :doc:`dynamic shapes guide<openvino_docs_OV_UG_DynamicShapes>`.
+Notes for performance and memory consumption in dynamic shape
+--------------------------------------------------------------
+
+- Extra CPU utilization during inference :
+
+   - Shape inference for new input shapes
+   - Kernel compilation in runtime for optimal kernel
+   - Unfusion of the fused subgraph when the fusing is illegal for runtime shape
+
+-  Higher memory consumption for in-memory cache
+
+   - Optimal kernels and weights from the previously used shapes are preserved in in-memory cache for future use
+
+
+Recommendations for performance improvement
+-----------------------------------------------------------
+
+- Use static shape whenever possible
+
+   - Static models can benefit from more agressive optimizations such as constant propagation, fusing, reorder optimization.
+   If the same shape is used for dynamic model, the performance is worse than running that shape with static model.
+   It is therefore recommended to reshape the dynamic model to a static model, if the scenario allows.
+
+- Use bounded dynamic shape whenever possible
+
+   - The GPU plugin needs to reallocate memory if the current shape is larger than the maximum of the previous shapes, which causes additional overhead.
+   - Using bounded dynamic shape will help reduce such overhead. For example, use {ov::Dimension(1, 10), ov::Dimension(1, 384)} instead of {ov::Dimension(-1), ov::Dimension(-1)}.
+   - Note that bounded dynamic *batch* is handled differently as mentioned above.
+
+- Use permanent cache, i.e., either OpenCL cache or OpenVino model_cache, to reduce runtime re-compilation overhead
+
+   - GPU plugin deploys in-memory cache to store compiled kernels for previously used shapes, but the size of such in-memory cache is limited.
+   Therefore, it is recommended to use either of the following permanent caches too:
+
+      - OpenVino model_cache: See the :doc:`Model caching overview<openvino_docs_OV_UG_Model_caching_overview>`. Note that this option is preferable for discrete GPUs, as it provides caching for OneDNN primitives not only for OpenCL primitives. 
+      - OpenCL cache : See the page : https://github.com/intel/compute-runtime/blob/master/opencl/doc/FAQ.md#feature-cl_cache
```

Comment:
what is more preferable in general between opencl cache and model_cache? I think it would be better to be clear on that.(even better if we can just say "use model_cache please")

Reply:
For iGPU, it does not matter (because currently dynamic shape model does not do more things in model shape). In the future model caching will be better, once it caches more for dynamic shape model.

Reply:
that what about just mention model caching alone? To my understanding, opencl caching is less preferable from OV user perspective because it is device specific option.(only for GPU, not for CPU)

Reply:
I think even now dynamic model export allows to skip transformations pipeline, so models caching seems to be a preferrable option.

Reply:
Removed cl_cache_dir relevant statements.